### PR TITLE
Refactor/custom components

### PR DESCRIPTION
**Pull request by hellais**

### CODE REVIEW

**Comment by sarathms** [comment]
@hellais I have added tests for all components
- Discovered a bug in `Modal`  #43 
- There are some minor problems in how InputWithIconButton is implemented. E.g Some props need not be passed on to all children.
- Some earlier tests were written without understanding snapshot testing fully, but are functional. Wondering if there is any point keeping them.

File: components/organisms/Modal.js [hunk 1]
```diff
@@ -70,10 +70,10 @@ export const Modal = (props) => {
   return (
     <StyledModal show={props.show}>
       <StyledModalContent {...props}>
-      {props.closeButton && <StyledCloseButton onClick={props.onHideClick} icon={<MdClose/>} position={props.closeButton} />}
+      {props.closeButton && <StyledCloseButton onClick={props.onCloseClick} icon={<MdClose/>} position={props.closeButton} />}
         {React.Children.map(props.children, child => React.cloneElement(child), props)}
       </StyledModalContent>
-      {props.show && <StyledModalBackdrop onClick={props.onHideClick} />}
+      {props.show && <StyledModalBackdrop onClick={props.onCloseClick} />}
```
Comment (hellais) [inline components/organisms/Modal.js]: It seems like in ooni/run we are using the prop `onHideClick`: https://github.com/ooni/run/blob/master/pages/index.js#L311. I don't have an opinion on if we should use the prop `onHideClick` or `onCloseClick` as long as when we cut a release with this we don't forget to update ooni/run to use the correct prop.

File: components/theme/colors.js [hunk 2]
```diff
@@ -15,153 +15,153 @@ let colors = {
...
+  'fuschia0': '#f8f0fc',
+  'fuschia1': '#f3d9fa',
+  'fuschia2': '#eebefa',
+  'fuschia3': '#e599f7',
+  'fuschia4': '#da77f2',
+  'fuschia5': '#cc5de8',
+  'fuschia6': '#be4bdb',
+  'fuschia7': '#ae3ec9',
+  'fuschia8': '#9c36b5',
+  'fuschia9': '#862e9c',
```
Comment (hellais) [inline components/theme/colors.js]: We should probably eventually just drop, but only after have done a check to ensure it's not referenced anywhere (this is an artifact from a typo in spelling of the color [fuchsia](https://en.wikipedia.org/wiki/Fuchsia_(color)))

File: components/organisms/Modal.js (hunk 1 above)
Comment (sarathms) [reply components/organisms/Modal.js]: Well, it looked like `onHideClick` was partially renamed to `onCloseClick`. If it is being used, then I will just update everything to `onHideClick` here.

File: components/theme/colors.js (hunk 2 above)
Comment (sarathms) [reply components/theme/colors.js]: We are not the only ones who made this mistake. There are 35K results for `fuschia` on Github. And here's [the reason](https://en.wikipedia.org/wiki/Fuchsia#Pronunciation_and_spelling) why:

> the word's German origin is "fook-sya" /ˈfʊksija/, the standard pronunciation for the common name in English is "fyu-sha" /ˈfjuːʃə/. As a consequence, fuchsia is often misspelled as fuschia in English.

Comment (sarathms) [reply components/theme/colors.js]: `fuschia` is only referenced in two repos, once each. In [explorer](https://github.com/ooni/explorer/blob/master/components/test-info.js#L26) and [probe-desktop](https://github.com/ooni/probe-desktop/blob/master/renderer/components/nettests/performance/index.js#L14).

If we drop these in this version, it breaks compatibility and hence major version upgrade?

File: components/organisms/Modal.js (hunk 1 above)
Comment (sarathms) [reply components/organisms/Modal.js]: This is done.

File: components/theme/colors.js (hunk 2 above)
Comment (sarathms) [reply components/theme/colors.js]: Never mind. We will do this separately. Ticket opened #44

**Pull request merged**